File: Source/Fantasy.Jobs/ReservedPropertiesTagProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.IO;
using System.Reflection;
using Fantasy.Jobs.Management;

namespace Fantasy.Jobs
{
    public class ReservedPropertiesTagProvider: ObjectWithSite, ITagValueProvider
    {
        #region ITagValueProvider Members

        public char Prefix
        {
            get { return '$'; }
        }


        private IJobEngine _engine;

        private IJobEngine Engine
        {
            get 
            {
                if (_engine == null)
                {
                    _engine = this.Site.GetRequiredService<IJobEngine>();
                    
                }
                return _engine;
            }
        }

        


        public string GetTagValue(string tag, IDictionary<string, object> context)
        {
            switch (tag.ToLower())
            {
                case "jobdir":
                    return this.Engine.JobDirectory;
                case "intermediatedir":
                    return this.Engine.JobDirectory;
                case "jobhostdir":
                    return LongPath.GetDirectoryName(Assembly.GetEntryAssembly().Location);
                case "templatedir":
                    return this.Site.GetRequiredService<IJobManagerSettingsReader>().GetSetting<string>("JobTemplateDirectoryFullPath");
                case "jobid":
                    return this.Engine.JobId.ToString();
                case "template":
                    IJob job = this.Site.GetRequiredService<IJob>();
                    return job.TemplateName;
                default:
                    return "";
            }
        }

        public bool HasTag(string tag, IDictionary<string, object> context)
        {
            string[] tags = new string[] { "jobdir", "intermediatedir", "jobhostdir", "templatedir", "jobid", "templatename"};
            return Array.IndexOf(tags, tag.ToLower()) >= 0;
        }

        public bool IsEnabled(IDictionary<string, object> context)
        {
            return true;
        }

        #endregion
    }
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/ReservedPropertiesTagProvider.java
package fantasy.jobs;

import java.util.*;

import fantasy.jobs.management.*;
import fantasy.*;

public class ReservedPropertiesTagProvider extends ObjectWithSite implements ITagValueProvider
{

	public final char getPrefix()
	{
		return '$';
	}


	private IJobEngine _engine;

	private IJobEngine getEngine() throws Exception
	{
		if (_engine == null)
		{
			_engine = this.getSite().getRequiredService(IJobEngine.class);

		}
		return _engine;
	}




	public final String GetTagValue(String tag, java.util.Map<String, Object> context) throws Exception
	{

		if (tag.toLowerCase().equals("jobdir"))
		{
				return this.getEngine().getJobDirectory();
		}
		else if (tag.toLowerCase().equals("intermediatedir"))
		{
				return this.getEngine().getJobDirectory();
		}
		else if (tag.toLowerCase().equals("jobhostdir"))
		{
				return JavaLibraryUtils.getEntryLibrary().getParent();
		}
		else if (tag.toLowerCase().equals("templatedir"))
		{
				return this.getSite().getRequiredService(IJobManagerSettingsReader.class).getSetting(String.class, "JobTemplateDirectoryFullPath");
		}
		else if (tag.toLowerCase().equals("jobid"))
		{
				return this.getEngine().getJobId().toString();
		}
		else if (tag.toLowerCase().equals("template"))
		{
				IJob job = this.getSite().getRequiredService(IJob.class);
				return job.getTemplateName();
		}
		else
		{
				return "";
		}
	}

	public final boolean HasTag(String tag, java.util.Map<String, Object> context)
	{
		String[] tags = new String[] { "jobdir", "intermediatedir", "jobhostdir", "templatedir", "jobid", "templatename"};
		return Arrays.asList(tags).indexOf(tag.toLowerCase()) >= 0;
	}

	public final boolean IsEnabled(java.util.Map<String, Object> context)
	{
		return true;
	}
}